radsat-sk/operation/subsystems/RDosimeter.c

```diff
@@ -134,22 +140,45 @@ int dosimeterCollectData(void)
 								dataResponse, DOSIMETER_I2C_DELAY);
 
 			// check for success of I2C command
-			if ( error != 0 )
+			if (error != 0)
 				return error;
 
 			float finalVoltage = convertCountsToVoltage(dataResponse[0], dataResponse[1]);
 
 			// if reading the temperature sensor, convert it to celsius
-			if (adcChannel == temperatureSensor) {
-				uint16_t temperature = convertVoltageToTemperature(finalVoltage);
-			}
+			if (adcChannel == temperatureSensor)
+				finalVoltage = convertVoltageToTemperature(finalVoltage);
 
-			// TODO: format data into Protobuf message
+			// store result in 2D array
+			results[dosimeterBoard][adcChannel] = finalVoltage;
 
 		}
 	}
 
-	// TODO: send formatted protobuf messages to downlink manager
+	// format protobuf message with recorded values
+
+	// board one
+	data.boardOne.voltageChannelZero = results[dosimeterBoardOne][adcChannelZero];
+	data.boardOne.voltageChannelOne = results[dosimeterBoardOne][adcChannelOne];
+	data.boardOne.voltageChannelTwo = results[dosimeterBoardOne][adcChannelTwo];
+	data.boardOne.voltageChannelThree = results[dosimeterBoardOne][adcChannelThree];
+	data.boardOne.voltageChannelFour = results[dosimeterBoardOne][adcChannelFour];
+	data.boardOne.voltageChannelFive = results[dosimeterBoardOne][adcChannelFive];
+	data.boardOne.voltageChannelSix = results[dosimeterBoardOne][adcChannelSix];
+	data.boardOne.voltageChannelSeven = results[dosimeterBoardOne][adcChannelSeven];
+
+	// board two
+	data.boardTwo.voltageChannelZero = results[dosimeterBoardTwo][adcChannelZero];
+	data.boardTwo.voltageChannelOne = results[dosimeterBoardTwo][adcChannelOne];
+	data.boardTwo.voltageChannelTwo = results[dosimeterBoardTwo][adcChannelTwo];
+	data.boardTwo.voltageChannelThree = results[dosimeterBoardTwo][adcChannelThree];
+	data.boardTwo.voltageChannelFour = results[dosimeterBoardTwo][adcChannelFour];
+	data.boardTwo.voltageChannelFive = results[dosimeterBoardTwo][adcChannelFive];
+	data.boardTwo.voltageChannelSix = results[dosimeterBoardTwo][adcChannelSix];
+	data.boardTwo.voltageChannelSeven = results[dosimeterBoardTwo][adcChannelSeven];
+
+	// send formatted protobuf messages to downlink manager
+	fileTransferAddMessage(&data, sizeof(data), FileTransferMessage_dosimeterData_tag);

```

Comment:
Just to confirm, in each subsystem module are we to prepare the telemetry, via the file transfer service, before sending it off? Additionally, in your block diagram the dosimeter board module does not connect to the file transfer service bus.

Reply:
Well that's actually up to us to decide, we have 2 options:
1. Modules can directly format and send their data to the File Transfer Service
    - The functions that do this get called by the FreeRTOS tasks
    - This is how it currently is designed
2. FreeRTOS Tasks can call the subsystem modules and get the data as a return (e.g. into a pointer), and then the Services themselves can send that data to the File Transfer Service

Personally I like the first option, it's a bit cleaner.